### PR TITLE
Prepare Makefile and spec file to use rpmbuild successfully

### PR DESCRIPTION
**Pull request by gdha**
We can now build an ebiso RPM package on SLES (and RHEL) based systems. For SLES I've uploaded the tar.gz and spec file to my OBS home project folders and builded these for all SLES/OpenSuSe related distro's, e.g.
http://download.opensuse.org/repositories/home:/gdha/SLE_11_SP3/x86_64/

### CODE REVIEW

File: packaging/ebiso.spec [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+%define rpmrelease  %nil
+
+
+### Work-around the fact that OpenSUSE/SLES _always_ defined both :-/
+%if 0%{?sles_version} == 0
+%undefine sles_version
+%endif
+
+Name:			ebiso
+Version:		0.1.1
```
Comment (gozora) [inline packaging/ebiso.spec]: @gdha can you please replace _0.1.1_ with `%{version}` here?

**Comment by gozora** [comment]
Hello @gdha, despite I'm not Makefile expert this one looks good to me. Just give me some time to run couple of tests ...

Thanks for your work!

File: packaging/ebiso.spec (hunk 1 above)
Comment (gdha) [inline packaging/ebiso.spec]: @gozora that would not work I am afraid as version would not be defined at this point. What I use to do is edit the spec file via the Makefile (via a rewrite rule - see the rear Makfile for an example).

File: Makefile [hunk 2]
```diff
@@ -39,28 +49,40 @@ ${PROGNAME}: ${OBJ} ${PROGNAME}.c ${HEADERS}
...
+		--define "_rpmfilename %%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm" \
+		--define "debug_package %{nil}" \
+		--define "_rpmdir %(pwd)" \
+		-tb  $(PROGNAME)-$(distversion).tar.gz
+
 .PHONY: install
 install: ${PROGNAME}
-	install -m 0755 ${PROGNAME} ${INSTDIR}
+	@echo -e "\033[1m== Installing $(PROGNAME)-$(distversion) ==\033[0;0m"
+	[ ! -d $(DESTDIR)$(INSTDIR) ] && mkdir -m 755 -p $(DESTDIR)$(INSTDIR)
```
Comment (gozora) [inline Makefile]: @gdha For some strange reason I get error for following piece of code: 
`[ ! -d $(DESTDIR)$(INSTDIR) ] && mkdir -m 755 -p $(DESTDIR)$(INSTDIR)`

```
uefi:(/tmp/gdha/ebiso)(root)# make install
== Installing ebiso-0.1.1 ==
[ ! -d /usr/bin ] && mkdir -m 755 -p /usr/bin
make: *** [install] Error 1
```

When I use this  instead all seems to be OK:
`if [ ! -d $(DESTDIR)$(INSTDIR) ]; then mkdir -m 755 -p $(DESTDIR)$(INSTDIR); fi`

```
uefi:(/tmp/gdha/ebiso)(root)# make install
== Installing ebiso-0.1.1 ==
if [ ! -d /usr/bin ]; then mkdir -m 755 -p /usr/bin; fi
install -m 0755 ebiso /usr/bin
```

Any clue what can be wrong?

I'm using make-3.81-128.20 ...

**Comment by gozora** [comment]
@gdha I've made some small changes in your version of Makefile:

```
--- ../gdha/orig/ebiso/Makefile 2015-10-26 17:39:20.000000000 +0100
+++ Makefile    2015-10-26 18:59:08.000000000 +0100
@@ -51,12 +51,13 @@ ${PROGNAME}: ${OBJ} ${PROGNAME}.c ${HEAD
 .PHONY: dist
 dist: 
    @echo -e "\033[1m== Creating tar archive $(PROGNAME)-$(distversion).tar.gz ==\033[0;0m"
+   make rewrite
    make clean
    tar czf ../$(PROGNAME)-$(distversion).tar.gz \
    --exclude=.git \
    --exclude=.gitignore \
    --exclude=README.md \
-   --transform='s,^\./,ebiso-0.1.1/,S' .
+   --transform='s,^\./,ebiso-$(distversion)/,S' .
    @mv ../$(PROGNAME)-$(distversion).tar.gz ./

 .PHONY: clean
@@ -64,6 +65,7 @@ clean:
    rm -f ${LIBDIR}/*.o ${LIBDIR}/*.a
    rm -f ${PROGNAME}
    rm -f $(PROGNAME)*.tar.gz
+   rm -f $(PROGNAME)*.rpm
    rm -rf ${BASE_DEPDIR}

 .PHONY: rpm
@@ -75,10 +77,15 @@ rpm: dist $(specfile)
        --define "_rpmdir %(pwd)" \
        -tb  $(PROGNAME)-$(distversion).tar.gz

+.PHONY: rewrite
+rewrite: 
+   @echo -e "\033[1m== Rewriting $(specfile) ==\033[0;0m"
+   sed -i.orig -e 's#^Version:.*#Version:\t\t$(distversion)#' $(specfile)
+
 .PHONY: install
 install: ${PROGNAME}
    @echo -e "\033[1m== Installing $(PROGNAME)-$(distversion) ==\033[0;0m"
-   [ ! -d $(DESTDIR)$(INSTDIR) ] && mkdir -m 755 -p $(DESTDIR)$(INSTDIR)
+   if [ ! -d $(DESTDIR)$(INSTDIR) ]; then mkdir -m 755 -p $(DESTDIR)$(INSTDIR); fi
    install -m 0755 ${PROGNAME} $(DESTDIR)$(INSTDIR)

 .PHONY: uninstall
```

Do you think we can use this version?

Thanks for that hint with rewrite, it is really usefull!

**Comment by gozora** [comment]
Finally I've managed  to (partially) understand OBS project :-/.
Since next commit ebiso packages will be available at my [OBS](http://download.opensuse.org/repositories/home:/gozora/SLE_11_SP3/) page.

**Comment by gdha** [comment]
@gozora Indeed the rewrite and restore rules are a step forward. I removed the SLES directory to get tid of the rpms and different spec files. OBS and rpmbuild do not like more then 1 spec file.
I also  `strip` the ebiso executable to get it smaller.
The package is almost good now:

```
$ rpmlint -vi ebiso-0.1.4-1.el7.centos.x86_64.rpm
ebiso.x86_64: I: checking
ebiso.x86_64: W: no-version-in-last-changelog
The latest changelog entry doesn't contain a version. Please insert the
version that is coherent with the version of the package and rebuild it.

ebiso.x86_64: I: checking-url https://github.com/gozora/ebiso (timeout 10 seconds)
ebiso.x86_64: W: no-documentation
The package contains no documentation (README, doc, etc). You have to include
documentation files.

ebiso.x86_64: W: no-manual-page-for-binary ebiso
Each executable in standard binary directories should have a man page.

1 packages and 0 specfiles checked; 0 errors, 3 warnings.
```

If you could provide a man page then we could propose this package to SuSe as they miss this feature for UEFI (and their is a demand for it).
Thanks for ebiso - I like it.
Gratien

**Comment by gozora** [comment]
@gdha No problem removing those dirs with RPMs and specfiles, I wanted to get rid of them anyway ...
I'll create man page today or tomorrow (hopefully) and will drop you a message.
Thanks again for your contribution and help, I really appreciate that.

Vlado

**Comment by gozora** [comment]
Hello @gdha 

Manual page was created/pushed.

Except that I had to change following code in Makefile as well

```
-   --transform='s,^\./,ebiso-$(distversion)/,S' .
+   --transform='s,^\.,ebiso-$(distversion),S' .
```

 because created tar archive contained entry for current directory (**./**)

```
uefi:(/tmp)(root)# tar tvf /share/ebiso-0.1.4.tar.gz
drwxr-xr-x 1001/1001     0 2015-10-27 22:14 ./
-rw-rw-r-- 1001/1001  3279 2015-10-27 22:14 ebiso-0.1.4/Makefile
drwxr-xr-x 1001/1001     0 2015-10-26 22:21 ebiso-0.1.4/include/
...
```

Wich obviously caused to change permissions/ownership of directory I've made extraction in.

**Pull request merged**